bc_obps/service/operation_service_v2.py

```diff
@@ -396,3 +397,22 @@ def check_conditions() -> Generator[Tuple[Callable[[], bool], str], None, None]:
         for condition, error_message in check_conditions():
             if not condition():
                 raise Exception(error_message)
+
+    @classmethod
+    def generate_boro_id(cls, user_guid: UUID, operation_id: UUID) -> BcObpsRegulatedOperation | None:
+        operation = OperationService.get_if_authorized(user_guid, operation_id)
+        is_eio = operation.registration_purposes.all().filter(
+            registration_purpose=RegistrationPurpose.Purposes.ELECTRICITY_IMPORT_OPERATION
+        )
+        if operation.bc_obps_regulated_operation:
+            raise Exception('Operation already has a BORO ID.')
+        if is_eio:
+            raise Exception('EIOs cannot be issued BORO IDs.')
+        if operation.status != Operation.Statuses.REGISTERED:
+            raise Exception('Operations must be registered before they can be issued a BORO ID.')
+
+        operation.generate_unique_boro_id()
+        operation.save(update_fields=['bc_obps_regulated_operation'])
+        # brianna do we want to do this, or better to set the verification columns?
```

Comment:
Do we want to set_create_or_update, or the verification columns? Verification columns were set in reg1 but not sure if that still applies

Reply:
excellent point. Since the "Registered" status is now set automatically, the verification columns on the operation aren't relevant in Reg2. However, for the `BC Obps Regulated Operation` model, we're only tracking `issued_at` and not `issued_by`. IMO we should add an `issued_by` column to this model so that we can track which internal user issued the BORO id. But I'll leave it up to you @BCerki to decide whether you want to include that as part of this PR, or we can create a separate ticket for that work to be done later.

Reply:
I can include it here, it'll be quick. So only set issued by, not verified or updated?

Reply:
cool, thanks Bri! We'll have a nice separation of audit columns this way actually - the `Operation` model's created and updated fields will only be affected by external users, and the `BcObpsRegulatedOperation` model will have `issued_by` and `issued_at` columns that are only triggered by internal users.

so yes, for this user story, an internal user issuing a BORO ID should only change the audit cols on the `BORO` model, not the Operation model.